packages/dart_mappable_builder/lib/src/generators/enum_mapper_generator.dart

```diff
@@ -41,9 +43,9 @@ class EnumMapperGenerator extends MapperGenerator<TargetEnumMapperElement> {
         }
         
         extension ${element.mapperName}Extension on ${element.prefixedClassName} {
-          ${element.hasAllStringValues ? 'String' : 'dynamic'} toValue() {
+          ${extensionReturnType} toValue() {
             ${element.mapperName}.ensureInitialized();
-            return MapperContainer.globals.toValue(this)${element.hasAllStringValues ? ' as String' : ''};
+            return MapperContainer.globals.toValue<${extensionReturnType}>(this)${element.hasAllStringValues ? ' as String' : ''};
```

Comment:
The T in toValue<T> should denote the input type, not the output type. So instead <dynamic> will suffice in all cases.